CMakeLists.txt

```diff
@@ -904,8 +904,10 @@ endif()
 # TODO: Can't deactivate on FreeBSD? w/o LIBC, SDL_stdinc.h can't define
 # anything.
 if(SDL_LIBC)
+  set(HAVE_LIBC TRUE)
+endif()
+if(SDL_LIBC OR SDL_STATIC)
```

Comment:
SDL_LIBC isn't already set to 1 when SDL_STATIC is true on WIN32?

Reply:
My current intention is to apply `SDL_LIBC` only on the shared library, and let the static library always use the system C library.
Though, this currently breaks all MSVC configurations.

Alternatively, the behavior from current master is kept and both shared and static libraries are built with the same `SDL_LIBC` property.
In that case, it is the responsibility of the consumer to build a static library with the system C library.
Keeping the current behavior means that DevilutionX would be unable to build against a SDL2 binary package shipping a static library built without the system C library.

Reply:
Sorry, I misunderstood, I thought SDL_LIBC and HAVE_LIBC were the same thing. Yes, shared library uses SDL builtin C runtime and static library uses system C library is correct behavior.

Reply:
I guarded the `process.h` header behind a `#if defined(HAVE_LIBC)`.
I did this initially because the `process.h` header defines `M_PI`, which clashes with `M_PI`, defined in `SDL_stdinc.h`.

Looking at the documentation of [`_beginthreadex`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/beginthread-beginthreadex) and [`_endthreadex`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/endthread-endthreadex), these are only available through the C runtime libraries.

Reply:
I'm thinking about an alternative strategy. I'm creating new pr's for all commits that can stand on their own.